src/common/mem.c

```diff
@@ -1149,10 +1254,19 @@ set_offsets(struct umem_instance *umm)
 
 		umm->umm_pool_uuid_lo = root_oid.pool_uuid_lo;
 		umm->umm_base = (uint64_t)root - root_oid.off;
-	} else {
-		dav_obj_t *pop = (dav_obj_t *)umm->umm_pool->up_priv;
+		break;
+	case UMEM_CLASS_BMEM:
+		dav_pop = (dav_obj_t *)umm->umm_pool->up_priv;
 
-		umm->umm_base = (uint64_t)dav_get_base_ptr(pop);
+		umm->umm_base = (uint64_t)dav_get_base_ptr(dav_pop);
+		break;
+	case UMEM_CLASS_ADMEM:
+		bh.bh_blob = (struct ad_blob *)umm->umm_pool->up_priv;
+		umm->umm_base = (uint64_t)ad_get_base_ptr(bh);
+		break;
+	default:
+		D_ASSERTF(0, "bad umm->umm_id %d\n", umm->umm_id);
```

Comment:
so mmap + set base will be implemented in follow-on patches?

Reply:
right, I'll add a few code later. thx